samples/dfu/README.rst

```diff
@@ -7,44 +7,70 @@ Overview
 This DFU application demonstrates how to connect with Golioth and use Device
 Firmware Upgrade (DFU) procedure.
 
+The app is first built and flashed to the device which will then listen to
+Golioth Cloud for a new firmware binary to become available. A second binary is
+built with a new version number, registered as an artifact, and depolyed as a
+release. The device will detect this change and automatically download and
+install.
+
```

Comment:
This is an improvement of documentation and is unrelated to "feat: add support for Thingy:91 to dfu sample" commit title. One logical change == one commit. This commit should result at least with following commits:
- `samples: dfu: fix dfu artifact and release commands`
- some other documantation fixes
- `samples: dfu: improve high-level sample description`
- `samples: dfu: add support for Thingy:91 board`

Please remember to first push fixes, then improvements. Preferably bugfixes and/or documentation improvements should be in a separate PR, because they are not related directly to inclusion of Thingy:91 board and having everything in single PR will just block bugfixes from being included.